src/openfermion/measurements/_bayesian_estimators.py

```diff
@@ -0,0 +1,796 @@
+                 num_vectors=1,
+                 num_freqs=1000,
+                 max_n=1,
+                 vector_guess=None,
+                 full_update_with_failure=False,
+                 store_history=True,
+                 amplitude_approx_cutoff=100):
+
+        '''
+        Args:
+            amplitude_guess: estimates for amplitudes of different
+                eigenstates in the initial state ($A_j$)
+            amplitude_vars: variance on the amplitude estimates
+            num_vectors: number of phases to estimate
+            num_freqs: number of frequencies in Fourier representation.
+            max_n: maximum number of unitary evolutions before measurement.
+                dictates how many matrices for updates are made and stored.
+            vector_guess: a prior estimate of the phases. If none,
+                assumes flat.
+            full_update_with_failure: chooses whether to perform
+                a full single step of SLSQP whenever the amplitude
+                optimization returns an unphysical result (negative
+                amplitude squared), or to just enforce physicality
+                by hand.
+            store_history: whether to store the history of
+                estimated values of the estimator. This is only
+                a few numbers per update, but it could be costly.
+            amplitude_approx_cutoff: a cutoff between estimating the amplitudes
+                via full SLSQP and approximating with single steps of Newton's
+                method. Increasing may lead to higher accuracy,
+                decreasing will lead to faster runtimes.
+        '''
+        self._amplitude_approx_cutoff = amplitude_approx_cutoff
+        self._full_update_with_failure = full_update_with_failure
+        self._store_history = store_history
+
+        super(BayesEstimator, self).__init__(
+            amplitude_guess=amplitude_guess,
+            amplitude_vars=amplitude_vars,
+            num_vectors=num_vectors,
+            num_freqs=num_freqs,
+            max_n=max_n,
+            vector_guess=vector_guess)
+
+    def reset(self):
+        '''
+        Resets estimator to initial state.
+        '''
+
+        super(BayesEstimator, self).reset()
+
+        # The following stores the history of the estimator
+        # for analysis purposes
+
+        if self._store_history:
+            self.averages = []
+            self.variances = []
+            self.log_bayes_factor_history = []
+            self.amplitudes_history = []
+
+        self.log_bayes_factor = 0
+        self.num_dsets = 0
+
+    def update(self, experiment_data, force_accept=False):
+        '''
+        Performs the expected Bayesian update, and stores result.
+
+        Args:
+            experiment_data: a list of dictionaries, containing
+                the data from each round of the experiment. Each round requires
+                entries: 'num_rotations', 'measurement', and 'final_rotation'.
```

Comment:
If `experiment_data` is only ever expected to have these three keys, then using a dictionary is overkill. Just use a tuple or collections.namedtuple instead.

Reply:
Experiment_data may also have a fourth key ('true_measurement'), and I can imagine adding other keys later on. I figured using a dict makes compatibility/usability easier, if you'd prefer I can change it though.

Reply:
Make a class to represent this experiment data concept. Instead of passing dictionaries, pass instances of that class.

Reply:
Alright, I've put something together. I've split it into a new file as these can be used for the time-series estimator as well.